api/src/utils/apply-query.ts

```diff
@@ -539,23 +539,23 @@ export function applyFilter(
 			// Knex supports "raw" in the columnName parameter, but isn't typed as such. Too bad..
 			// See https://github.com/knex/knex/issues/4518 @TODO remove as any once knex is updated
 
-			// These operators don't rely on a value, and can thus be used without one (eg `?filter[field][_null]`)
-			if (operator === '_null' || (operator === '_nnull' && compareValue === false)) {
-				dbQuery[logical].whereNull(selectionRaw);
+			const boolValue = Boolean(compareValue ?? true);
```

Comment:
I think there's no need to extract this `boolValue`, this leads to the conditions using it being less verbose. The conditions simply checks if `compareValue` is `false` strictly, which would invert the operation. This change affects other falsy values such as `0`. I think the changes here should be reverted back to https://github.com/directus/directus/pull/16399/commits/4305332dd1b8d27e135854b89408d825602a84d6.